### DISCUSSION TITLE
Issues sending commands to panel - Vista20SE

### DISCUSSION THREAD

**Discussion by jakeiscrazy** [Dilbert66/esphome-vistaECP#118] [General]
Hey all, 

First of all thanks for all the excellent work on this project! 

I am successfully receiving messages from the alarm panel! However, I am not able to send any commands. I don’t get any errors in the logs. Suspect the issue is my opto isolator wiring (I’m a bit new to those). It is a 4n35.

Can someone take a look and see if there is anything that could be causing an issue here? 

I have combined a 10k and 22k resistor in
![IMG_4646](https://github.com/Dilbert66/esphome-vistaECP/assets/34757421/b1da23c2-3cae-4212-8534-2b06fcf59e36)
![IMG_4647](https://github.com/Dilbert66/esphome-vistaECP/assets/34757421/63e7f14b-7c1b-42cf-8c37-f8e0ddde6908)
 place of the 33k one because I didn’t have those on hand.

Thanks!

**Comment by Dilbert66** [comment]
FYI, this panel is different than the vista20.  It only works with this version: https://github.com/Dilbert66/esphome-vistaECP/tree/master/vista20SE .  Make sure that's what you are using.   Also it doesnt look like pin 2 of the opto is connected to ground.

**Reply by jakeiscrazy** [reply]
This was exactly the problem! I can't believe I didn't see the missing  ground leg! Everything now runs perfectly! Thank you again for this amazing project!
